control_data_collecting_tool/scripts/data_collecting_rosbag_record.py

```diff
@@ -0,0 +1,179 @@
+            if topic_type:
+                msg_module = get_message(topic_type)
+                sbuscriber_ = self.node.create_subscription(
+                    msg_module, topic_name, partial(self.callback_write_message, topic_name), 10
+                )
+                self.message_sbuscribers_.append(sbuscriber_)
+
+    # call back function called in start recording
+    def callback_write_message(self, topic_name, message):
+        try:
+            serialized_message = serialize_message(message)
+            current_time = rclpy.clock.Clock().now()
+            self.message_writer.write(topic_name, serialized_message, current_time.nanoseconds)
+        except Exception as e:
+            self.node.get_logger().error(f"Failed to write message: {e}")
+
+    def stop_record(self):
+        # Stop recording by destroying the subscriptions and deleting the writer
+        for sbuscriber_ in self.message_sbuscribers_:
+            self.node.destroy_subscription(sbuscriber_)
+        del self.message_writer
+
+
+class DataCollectingRosbagRecord(Node):
+    def __init__(self):
+        super().__init__("data_collecting_rosbag_record")
+        package_share_directory = get_package_share_directory("control_data_collecting_tool")
+        topic_file_path = os.path.join(package_share_directory, "config", "topics.yaml")
+        with open(topic_file_path, "r") as file:
+            topic_data = yaml.safe_load(file)
+        self.topics = topic_data["topics"]
+        self.writer = MessageWriter(self.topics, self)
+        self.subscribed = False
+        self.recording = False
+
+        self.present_operation_mode_ = None
+        self.operation_mode_subscriber_ = self.create_subscription(
+            OperationModeState,
+            "/system/operation_mode/state",
+            self.subscribe_operation_mode,
+            10,
+        )
+
+        self.timer_period_callback = 1.0
+        self.timer_callback = self.create_timer(
+            self.timer_period_callback,
+            self.record_message,
+        )
+
+    def subscribe_operation_mode(self, msg):
+        self.present_operation_mode_ = msg.mode
+
+    def record_message(self):
+        # Start subscribing to topics and recording if the operation mode is 3(LOCAL)
+        if self.present_operation_mode_ == 3 and not self.subscribed and not self.recording:
+            self.writer.create_writer()
+            self.writer.subscribe_topics()
+            self.subscribed = True
+
+        # Start recording if topics are subscribed and the operation mode is 3
+        if self.present_operation_mode_ == 3 and self.subscribed and not self.recording:
+            self.writer.start_record()
+            self.recording = True
+
+        # Stop recording if the operation mode changes from 3
+        if self.present_operation_mode_ != 3 and self.subscribed and self.recording:
+            self.writer.stop_record()
+            self.subscribed = False
+            self.recording = False
```

Comment:
please add message print in the terminal when start/stop record

Reply:
Thank you for your feedback. I have made the changes in [c95cdfb](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/c95cdfb26f5ff18858b7fb1a74dc769845b28b98).